### PR TITLE
add scripts for running lint within docker container; run lint via github actions

### PR DESCRIPTION
**Pull request by youngj**
Currently it seems that most frontend developers (including me) haven't been running `npm run lint` on their PRs. If one developer runs lint on their PR, lint will then make a lot of changes unrelated to their PR, which makes reviewing and merging difficult (e.g. https://github.com/trynmaps/metrics-mvp/pull/509).

However, in order for a developer to run `npm run lint`, they need npm installed on their host, but many developers do not have npm installed on their host, because normally npm runs inside a Docker container.

This PR adds scripts to allow developers using Docker to run lint inside a Docker container, so it is no longer necessary for developers to have npm installed on their host.

It also runs the lint task via Github Actions on each push so that we can see failing lint checks when reviewing PRs, and fixes all existing lint errors.

### CODE REVIEW

**Comment by youngj** [comment]
I just fixed the ~40 or so errors that were reported by `npm run lint`, so the checks will be passing after this is merged into master. This will create some merge conflicts in https://github.com/trynmaps/metrics-mvp/pull/528 which I'll fix once someone approves this.

File: frontend/.eslintrc.json [hunk 1]
```diff
@@ -24,11 +24,17 @@
         "ignoreUrls": true
       }
     ],
+    "no-param-reassign": [
+      "error",
+      {
+        "props": false
```
Comment (exxonvaldez) [inline frontend/.eslintrc.json]: Is this change needed?  If I understand the docs right  for this, isn't props: false the default?  https://eslint.org/docs/rules/no-param-reassign

Comment (youngj) [reply frontend/.eslintrc.json]: I noticed that the docs say props: false is the default, but the docs don't seem to be correct because lint was reporting several errors before adding this (https://github.com/trynmaps/metrics-mvp/runs/422180446): 

```
/app/frontend/src/components/ReactSelect.jsx
  427:9  error  Assignment to property of function parameter 'menuTransition'  no-param-reassign

/app/frontend/src/reducers/index.js
  22:7  error  Assignment to property of function parameter 'map'  no-param-reassign
  26:7  error  Assignment to property of function parameter 'map'  no-param-reassign

/app/frontend/src/routesMap.js
  43:11  error  Assignment to property of function parameter 'newDaysOfTheWeek'  no-param-reassign
  45:11  error  Assignment to property of function parameter 'newDaysOfTheWeek'  no-param-reassign
```

Comment (exxonvaldez) [reply frontend/.eslintrc.json]: Oh, I see... it's because we extend airbnb and they use props: true.  https://github.com/immerjs/immer/issues/189 .  Ok, so we definitively want this.

**Comment by youngj** [comment]
Sounds good, just added a section about the docker-lint scripts to README.md.
